src/QA/QA.Web/Controllers/BlockedOrganisationsController.cs

```diff
@@ -0,0 +1,126 @@
+using Microsoft.AspNetCore.Mvc;
+using Esfa.Recruit.Qa.Web.Configuration.Routing;
+using System.Threading.Tasks;
+using Esfa.Recruit.QA.Web.Orchestrators;
+using Esfa.Recruit.Qa.Web.Security;
+using Microsoft.AspNetCore.Authorization;
+using Esfa.Recruit.QA.Web.ViewModels.ManageProvider;
+using Esfa.Recruit.Qa.Web.Configuration;
+using Esfa.Recruit.Qa.Web.Extensions;
+
+namespace Esfa.Recruit.QA.Web.Controllers
+{
+    [Authorize(Policy = AuthorizationPolicyNames.TeamLeadUserPolicyName)]
+    [Route(RoutePaths.BlockedOrganisations)]
+    public class BlockedOrganisationsController : Controller
+    {
+        private readonly BlockedOrganisationsOrchestrator _orchestrator;
+        private readonly UserAuthorizationService _userAuthorizationService;
+        public BlockedOrganisationsController(UserAuthorizationService userAuthorizationService, BlockedOrganisationsOrchestrator orchestrator)
+        {
+            _orchestrator = orchestrator;
+            _userAuthorizationService = userAuthorizationService;
+        }
+
+        [HttpGet("", Name = RouteNames.BlockedOrganisations_Get)]
+        public async Task<IActionResult> Index()
+        {
+            var vm = await _orchestrator.GetBlockedOrganisationsViewModel();
+            return View(vm);
+        }
+
+        [HttpGet("find-training-provider", Name = RouteNames.FindTrainingProvider_Get)]
+        public IActionResult FindTrainingProvider()
+        {
+            if(TempData.ContainsKey(TempDataKeys.BlockProviderUkprnKey)) TempData.Remove(TempDataKeys.BlockProviderUkprnKey);
```

Comment:
Because I am peeking in some places, unless user complete the process, the TempData will be available to following requests, hence removing any stale data at the start of the process.